ext/standard/basic_functions.stub.php

```diff
@@ -6,7 +6,7 @@
  * @var float
  * @cname M_E
  */
-const M_E = 2.7182818284590452354;
+const M_E = 2.718281828459045;
```

Comment:
why is `M_E` declared here and no other math constants?

Reply:
Oh, indeed, that looks fishy. [We use the definitions provided by libc, if available, and only define the macros ourselves otherwise](https://github.com/php/php-src/blob/eff9aed1592f59cddb12d36a55dec0ccc3bbbfd6/ext/standard/php_math.h#L31-L97). There is nothing special about `M_E` compared to the other constants (like `M_PI`).

However, given that we already go to great lengths to have portable floating point arithmetic, it may not be the best idea to rely on definitions provided by libc.